SwanFileBrowser/src/swanfilebrowser.ts

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) SWAN Development Team.
+// Author: Omar.Zapata@cern.ch 2021
+
+/**
+ * This file containts the implementation for SwanFileBrowser and SwanFileBrowserModel classes.
+ *
+ * Those classes are the replacement for the default FileBrowser and allows to
+ * manipulated the paths to check if the folder is a project or not and to set the proper kernel path.
+ */
+
+import { FilterFileBrowserModel, FileBrowser } from '@jupyterlab/filebrowser';
+import { request } from './request';
+import { validateSpecModels } from './kernelspec/validate';
+import { SwanDirListing } from './listing';
+import { JSONObject } from '@lumino/coreutils';
+import { showErrorMessage, Dialog } from '@jupyterlab/apputils';
+import { CommandRegistry } from '@lumino/commands';
+/**
+ * SWAN Project options from .swanproject
+ * this is required to validate that the project file is not corrupted
+ */
+export interface ISwanProjectOptions {
+  name?: string;
+  stack?: string;
+  release?: string;
+  platform?: string;
+  user_script?: string;
+  python2?: any;
+  python3?: any;
+  kernel_dirs?: string[];
+}
+
+/**
+ * Customized SwanFileBrowserModel that inherits from FilterFileBrowserModel.
+ *
+ * This class has overloaded the method 'async cd(newValue: string): Promise<void>'
+ * that a allows to take actions before go to the directory, actions like:
+ * 1) Get the contents of the folder (to check if the folder is a project)
+ * 2) If the  folder is a project we can ge the information of the project stored in the .swanproject
+ * 3) if the information for the project is right then we can set the kernel spec manager
+ * 4) If the project is corrupted, the a Dialog is showed up telling the user the project requires
+ *    to be configured again and the ProjectDialog is called.
+ */
+export class SwanFileBrowserModel extends FilterFileBrowserModel {
+  constructor(
+    options: FilterFileBrowserModel.IOptions,
+    commads: CommandRegistry
+  ) {
+    super(options);
+    this._commands = commads;
+    this.kernelSpecSetPathRequest(this.path);
+  }
+
+  /**
+   * Request to set the kernelspec manager path in the backend.
+   * Local service manager.services.kernelspecs is updated as well.
+   *
+   * @param path path get information from jupyter api
```

Comment:
By "path get information" you mean "path to get information"?

Reply:
yes